src/main/java/duke/component/TaskList.java

```diff
@@ -0,0 +1,114 @@
+package duke.component;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a TaskList object that stores all task in a list.
+ * The 'TaskList' class supports operators (i) adding a task to list,
+ * (ii) deleting a task from list,
+ * (iii) getting a specific task from list,
+ * (iv) finding size of list, and
+ * (v) marking a task in list as done.
+ */
+public class TaskList {
+
+    /**
+     * List of tasks.
+     */
+    private ArrayList<Task> todoList;
```

Comment:
Just a small semantic nitpick but a todoList might imply a list of _todo_ tasks instead a list of tasks. I totally understand this however. 😅